### PR TITLE
Always attach CORS headers in REST

### PR DESCRIPTION
**Pull request by lutovich**
Previously CORS headers were only added to requests that passed auth. This made JS requests from browser unable to receive the actual error because CORS is not allowed with missing headers.

This PR moves the relevant filter to the top of the filter chain so that every response has CORS headers attached.

Fixes https://github.com/neo4j/neo4j/issues/9151

### CODE REVIEW

File: community/server/src/main/java/org/neo4j/server/rest/web/CorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * This filter adds the header "Access-Control-Allow-Origin : *" to all
+ * responses that goes through it. This allows modern browsers to do cross-site
+ * requests to us via javascript.
+ */
+public class CorsFilter implements Filter
+{
+    private static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    private static final String ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods";
+    private static final String ACCESS_CONTROL_ALLOW_HEADERS = "Access-Control-Allow-Headers";
+    private static final String ACCESS_CONTROL_REQUEST_METHOD = "access-control-request-method";
```
Comment (jimwebber) [inline community/server/src/main/java/org/neo4j/server/rest/web/CorsFilter.java]: capitalise to Access-Control-Request-Method

File: community/server/src/main/java/org/neo4j/server/rest/web/CorsFilter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    private static final String ACCESS_CONTROL_REQUEST_HEADERS = "access-control-request-headers";
```
Comment (jimwebber) [inline community/server/src/main/java/org/neo4j/server/rest/web/CorsFilter.java]: Capitalise to Access-Control-Request-Headers

**Review by jimwebber — CHANGES_REQUESTED** [review]
I think those headers are capitalised to be a good citizen in the HTTP world. Otherwise this seems sane to me.

**Review by ali-ince — APPROVED** [review]
LGTM!

I'm aware that this filter was already part of the product as it is here, but I think it may be important to discuss the logic of setting `Access-Control-Allow-Origin` to `*`  without a way to limit the cross-origin requests originators. The same logic applies to `Access-Control-Request-Method` and `Access-Control-Allow-Methods`. WDYT?

**Pull request merged**